Fix for issue 14348

adp-knaumann opened the pull request:
Self written modules in roles with become_user != root are broken

Fix: Also make the args file world readable.

DenkBrettl committed:
Fix for issue 14348

Self written modules in roles with become_user != root are broken

Fix: Also make the args file world readable.

jimi-c commented:
I believe there should be an `if args_file_path:` above that, as it may be `None`.

jimi-c commented:
Even better, if we could chmod both of these in one shot to save the round trip time.

DenkBrettl committed:
- only chmod the args file if it exists
- combine the chmod or module and args file into one command

DenkBrettl committed:
Actually the fix before doesn't work because due to quoting it's
not possible to chmod more than one file within one call as far
as I see it.
So now it works again.

jimi-c commented:
Squashed the above commits into one and merged:

<pre>
commit 1ea21f1f8ce3b9fa74a8803c4b87af50636445c6
Author: James Cammarata <jimi@sngx.net>
Date:   Mon Feb 8 10:00:26 2016 -0500
    Also chmod the args file path when using become for old-style modules
    
    Fixes #14348
</pre>

Also cherry-picked the above to `stable-2.0`.

Thanks!
